Headers/AppKit/NSLayoutConstraint.h

```diff
@@ -0,0 +1,180 @@
+
+// Orientation
+enum {
+  NSLayoutConstraintOrientationHorizontal = 0,
+  NSLayoutConstraintOrientationVertical = 1
+};
+typedef NSInteger NSLayoutConstraintOrientation;
+
+// Attributes
+enum {
+  NSLayoutAttributeLeft = 1,
+  NSLayoutAttributeRight,
+  NSLayoutAttributeTop,
+  NSLayoutAttributeBottom,
+  NSLayoutAttributeLeading,
+  NSLayoutAttributeTrailing,
+  NSLayoutAttributeWidth,
+  NSLayoutAttributeHeight,
+  NSLayoutAttributeCenterX,
+  NSLayoutAttributeCenterY,
+  NSLayoutAttributeLastBaseline,
+  NSLayoutAttributeBaseline = NSLayoutAttributeLastBaseline,
+  NSLayoutAttributeFirstBaseline, 
+  NSLayoutAttributeNotAnAttribute = 0
+};
+typedef NSInteger NSLayoutAttribute;
+  
+// Relation
+enum {
+  NSLayoutRelationLessThanOrEqual = -1,
+  NSLayoutRelationEqual = 0,
+  NSLayoutRelationGreaterThanOrEqual = 1,
+};
+typedef NSInteger NSLayoutRelation;
+  
+// Options
+enum {
+  NSLayoutFormatAlignAllLeft = (1 << NSLayoutAttributeLeft),
+  NSLayoutFormatAlignAllRight = (1 << NSLayoutAttributeRight),
+  NSLayoutFormatAlignAllTop = (1 << NSLayoutAttributeTop),
+  NSLayoutFormatAlignAllBottom = (1 << NSLayoutAttributeBottom),
+  NSLayoutFormatAlignAllLeading = (1 << NSLayoutAttributeLeading),
+  NSLayoutFormatAlignAllTrailing = (1 << NSLayoutAttributeTrailing),
+  NSLayoutFormatAlignAllCenterX = (1 << NSLayoutAttributeCenterX),
+  NSLayoutFormatAlignAllCenterY = (1 << NSLayoutAttributeCenterY),
+  NSLayoutFormatAlignAllLastBaseline = (1 << NSLayoutAttributeLastBaseline),
+  NSLayoutFormatAlignAllFirstBaseline = (1 << NSLayoutAttributeFirstBaseline),
+  NSLayoutFormatAlignAllBaseline = NSLayoutFormatAlignAllLastBaseline,
+  NSLayoutFormatAlignmentMask = 0xFFFF,
+  NSLayoutFormatDirectionLeadingToTrailing = 0 << 16, 
+  NSLayoutFormatDirectionLeftToRight = 1 << 16,
+  NSLayoutFormatDirectionRightToLeft = 2 << 16,    
+  NSLayoutFormatDirectionMask = 0x3 << 16,
+};
+typedef NSUInteger NSLayoutFormatOptions;
+
+@interface NSLayoutConstraint : NSObject <NSCoding, NSCopying>
+{
+  NSLayoutAnchor *_firstAnchor;
+  NSLayoutAnchor *_secondAnchor;
+  id _firstItem;
+  id _secondItem;
+  NSLayoutAttribute _firstAttribute;
+  NSLayoutAttribute _secondAttribute;
+  NSLayoutRelation _relation;
+  CGFloat _multiplier;
+  CGFloat _constant;
+  NSLayoutPriority _priority;
+}
+
++ (NSArray *)constraintsWithVisualFormat: (NSString *)fmt 
+                                 options: (NSLayoutFormatOptions)opt 
+                                 metrics: (NSDictionary *)metrics 
+                                   views: (NSDictionary *)views;
+
+// GNUstep extension...
+- (instancetype) initWithItem: (id)view1 
+                    attribute: (NSLayoutAttribute)attr1 
+                    relatedBy: (NSLayoutRelation)relation 
+                       toItem: (id)view2 
+                    attribute: (NSLayoutAttribute)attr2 
+                   multiplier: (CGFloat)mult 
+                     constant: (CGFloat)c;  
+
+// Designated initializer...
```

Comment:
This is a factory method not an initializer.

Reply:
True. I'll correct it.

Reply:
Done, will close when committed.